wfe/wfe.go

```diff
@@ -2565,3 +2587,119 @@ func (wfe *WebFrontEndImpl) processRevocation(
 	})
 	return nil
 }
+
+// Verify the External Account Binding in the request and return the key ID if requested and found.
+func (wfe *WebFrontEndImpl) verifyEAB(newAcctReq newAccountRequest, outerPostData *authenticatedPOST) (*string, *acme.ProblemDetails) {
+	if newAcctReq.ExternalAccountBinding == nil {
+		if wfe.requireEAB {
+			return nil, acme.ExternalAccountRequiredProblem(
+				"The request must include a value for the 'externalAccountBinding' field")
+		}
+
+		return nil, nil
+	}
+
+	eabBytes, err := json.Marshal(newAcctReq.ExternalAccountBinding)
+	if err != nil {
+		return nil, acme.MalformedProblem(err.Error())
+	}
+
+	//1.  Verify that the value of the field is a well-formed JWS
+	innerJWS, err := wfe.parseJWS(string(eabBytes))
+	if err != nil {
+		return nil, acme.MalformedProblem(err.Error())
+	}
+
+	//2.  Verify that the JWS protected field meets the following criteria
+	//-  The "alg" field MUST indicate a MAC-based algorithm
+	//-  The "kid" field MUST contain the key identifier provided by the CA
+	//-  The "nonce" field MUST NOT be present
+	//-  The "url" field MUST be set to the same value as the outer JWS
+	keyID, prob := wfe.verifyEABPayloadHeader(innerJWS, outerPostData)
+	if prob != nil {
+		return nil, prob
+	}
+
+	//3.  Retrieve the MAC key corresponding to the key identifier in the
+	//    "kid" field
+	key, ok := wfe.db.GetExtenalAccountKeyByKeyID(keyID)
+	if !ok {
+		return nil, acme.BadPublicKeyProblem(
+			"the field 'kid' references a key that does not exist")
+	}
+
+	keyBytes, err := base64.RawURLEncoding.DecodeString(key)
+	if err != nil {
+		return nil, acme.MalformedProblem("failed to decode CA key")
```

Comment:
```suggestion
		return nil, acme.InternalErrorProblem("failed to decode stored external account binding key")
```

Two things here:
1. I think the message should specifically say it's a stored external account binding key that failed to decode. "CA key" is probably too ambiguous.
2. I think this case would be a server internal problem. It indicates the Pebble instance was configured with an EAB key that it can't decode itself and that's not a problem with the client's request.